### PR TITLE
Fixed email validation on signup page

### LINKED ISSUE #1660: Email validation is not working on signup modal
**Actual Behaviour**

When typing the wrong email in the email textfield there is error shown.

**Expected Behaviour**

When typing the wrong email in the email textfield error should be shown.

**Screenshots of the issue**

email validation

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by samagragupta**
Fixes #1660 

Changes: 

- When typing the wrong email in the email textfield error is shown.
- On signing up using an invalid email address, Signup button will not work.

 

Demo Link: https://pr-1661-fossasia-susi-web-chat.surge.sh

Screenshots for the change: 
email validation

### CODE REVIEW

**Comment by samagragupta** [comment]
@akshatnitd I have done just similar to [accounts.susi.ai](https://accounts.susi.ai/signup) and [skills.susi.ai](https://skills.susi.ai/)

**Comment by aswinshenoy** [comment]
@akshatnitd @samagragupta  maybe we should show the error message only after the user focuses out of the email field and moves to the next field, and not while he is entering.

**Comment by AakashMallik** [comment]
@aswinshenoy Say the user focuses on the email input and without typing anything he moves on to the password field. In this case, your approach would end up giving an error message. Maybe the user wanted to input password first so he moved away from email. I think we should show error messages only when the user confirms that he is going to signup with the data he has provided. So my suggestion would be to link the signup button with validation.

**Comment by samagragupta** [comment]
@aswinmprabhu @AakashMallik I have just tried to make it consistence with [accounts.susi.ai](https://accounts.susi.ai/signup) and [skills.susi.ai](https://skills.susi.ai/)

**Comment by AakashMallik** [comment]
> @aswinmprabhu @AakashMallik I have just tried to make it consistence with [accounts.susi.ai](https://accounts.susi.ai/signup) and [skills.susi.ai](https://skills.susi.ai/)

We get that, we are just suggesting a better alternative. We are trying to improve the code, right? So why don't you implement it here and maybe someone else will copy your way on the accounts.susi.ai side. Just an opinion. @samagragupta

**Comment by samagragupta** [comment]
@AakashMallik @aswinmprabhu Done the required changes.
signup

**Comment by samagragupta** [comment]
@fragm3 Form validation on email text field is working fine. What more changes should I do?

**Comment by samagragupta** [comment]
@isuruAb Then what type of changes should I do?
Does the previous change in this PR is more correct?

File: src/components/Auth/SignUp/SignUp.react.js [hunk 1]
```diff
@@ -144,11 +144,12 @@ export default class SignUp extends Component {
     let state = this.state;
     if (event.target.name === 'email') {
       email = event.target.value.trim();
-      let validEmail = /^[A-Z0-9._%+-]+@[A-Z0-9.-]+\.[A-Z]{2,4}$/i.test(email);
```
Comment (Orbiter) [inline src/components/Auth/SignUp/SignUp.react.js]: I am not happy with the removal of the email verification. What is wrong with this regex? Please explain.

Comment (samagragupta) [reply src/components/Auth/SignUp/SignUp.react.js]: 1. 

email validation

2. 

signup

Which would be better?
In first one regex is used.

**Comment by AakashMallik** [comment]
@samagragupta I think the current implementation of Email validation has been finalized. This PR won't be needed. @akshatnitd @Akshat-Jain

**Pull request closed without merging**